### DISCUSSION TITLE
[Question]: How can I build a Q&A agent over a csv file present locally with Gradio UI?

### DISCUSSION THREAD

**Discussion by abhipn** [microsoft/autogen#1623] [Q&A] [answered]
I'm attempting to construct a multi-agent system where I can load a CSV into memory using pandas, and then ask questions about the data for the agents to discuss and reach a solution. I want to deploy this Q&A system using Gradio, but when I transfer the logic to a .py file, it seems like the DataFrame isn't retained in memory. I could prompt it to read the file every time the code is executed, but that introduces additional latency due to the file's size. Is there a better way to accomplish this? In LangChain, I have utilized a pandas agent to achieve similar tasks, but I couldn't find anything helpful here for solving this problem

FYI: In a Jupyter notebook I was able to make it work. 

### Additional Information

Python 3.10

**Comment by sonichi** [comment]
@ekzhu @olimoz

**Comment by ekzhu** [comment, marked as answer]
This is related to #261. See my answer: https://github.com/microsoft/autogen/discussions/261#discussioncomment-8431447

If you have a prototype or notebook example you would like to share, you are welcome to post it in our Discord channel `#agent-coding` and submit a PR.
